Add SAI Notification support for host_tx_ready

noaOrMlnx opened the pull request.

noaOrMlnx committed:
Add SAI Notification support for host_tx_ready

Junchao-Mellanox commented on meta/Meta.cpp:
```diff
@@ -6434,6 +6434,55 @@ void Meta::meta_sai_on_nat_event(
+            &sai_metadata_enum_sai_port_host_tx_ready_status_t,
+            host_tx_ready_status))
+    {
+        SWSS_LOG_WARN("port host_tx_ready value (%d) not found in sai_port_host_tx_ready_status_t",
```
do you want to return here and drop the notification is the value is invalid?

Junchao-Mellanox commented on meta/Meta.cpp:
```diff
@@ -6434,6 +6434,55 @@ void Meta::meta_sai_on_nat_event(
+    switch (ot)
+    {
+        case SAI_OBJECT_TYPE_PORT:
+        // case SAI_OBJECT_TYPE_BRIDGE_PORT:
```
remove unused comment

noaOrMlnx replied:
@Junchao-Mellanox 
I should expect to receive only SAI_OBJECT_TYPE_PORT right?

Junchao-Mellanox commented on meta/Meta.cpp:
```diff
@@ -6434,6 +6434,55 @@ void Meta::meta_sai_on_nat_event(
+    auto ot = objectTypeQuery(port_id);
+    auto valid = false;
+    
+    switch (ot)
```
do you need the switch/case here? how about if? 
```
valid =  (ot == SAI_OBJECT_TYPE_PORT);
```

Junchao-Mellanox commented on meta/Meta.cpp:
```diff
@@ -6434,6 +6434,55 @@ void Meta::meta_sai_on_nat_event(
+            break;
+    }
+
+    if (valid && !m_oids.objectReferenceExists(port_id))
```
could you please help me understand why `!m_oids.objectReferenceExists(port_id)` is needed here? My understanding is like this: 
```
if port ref does not exist:
     insert port ref

if host ready key does not exist:
    create host key

```

Junchao-Mellanox commented on meta/NotificationPortHostTxReadyEvent.cpp:
```diff
@@ -0,0 +1,73 @@
+
+using namespace sairedis;
+
+// NotificationSwitchStateChange::NotificationSwitchStateChange(
```
please remove unused comment

Junchao-Mellanox commented on meta/NotificationPortHostTxReadyEvent.cpp:
```diff
@@ -0,0 +1,73 @@
+{
+    SWSS_LOG_ENTER();
+
+    if (m_portId != SAI_NULL_OBJECT_ID)
```
no need the if branch, suggest just:
```
return m_portId;
```

Junchao-Mellanox commented on meta/SaiSerialize.cpp:
```diff
@@ -4593,6 +4639,31 @@ void sai_deserialize_port_oper_status_ntf(
+        SWSS_LOG_ERROR("Recieved an exception after trying to parse %s", s.c_str());
+        return;
+    }
+    // auto count = (uint32_t)j.size();
```
please remove unused comment

Junchao-Mellanox commented on meta/sai_serialize.h:
```diff
@@ -498,6 +513,9 @@ void sai_deserialize_free_bfd_session_state_ntf(
         _In_ uint32_t count,
         _In_ sai_bfd_session_state_notification_t* bfdsessionstate);
 
+void sai_deserialize_free_port_host_tx_ready_ntf(
```
do we really need it?

Junchao-Mellanox commented on syncd/scripts/syncd_init_common.sh:
```diff
@@ -233,6 +233,11 @@ config_syncd_mlnx()
         cat /tmp/sai_extra.profile >> /tmp/sai.profile
     fi
+
+    if [[ -f /$HWSKU_DIR/independent_mode_support.profile ]]; then
```
why need this?

noaOrMlnx replied:
got in here by mistake form another commit. removing

kcudnik committed:
Update SAI submodule to latest (#1311)

To include allow null on SAI_PORT_ATTR_PORT_SERDES_ID

Ashish1805 committed:
[Link event damping] Add generic concurrent queue for link event damping. (#1297)

This queue will be used to enqueue the port state change events by NotificationHandler and dequeued by syncd main thread and processed in link event damping logic.

HLD: sonic-net/SONiC#1071

Ashish1805 committed:
[Link event damping] Add utility methods. (#1313)

- Adding utility methods to calculate:
  1) Remaining value of a substance after time t using half-life
     formula.
  2) Time to reach a target value using half-life formula.

HLD: sonic-net/SONiC#1071

stephenxs commented on meta/NotificationFactory.cpp:
```diff
@@ -6,6 +6,7 @@
 #include "NotificationSwitchShutdownRequest.h"
 #include "NotificationSwitchStateChange.h"
 #include "NotificationBfdSessionStateChange.h"
+#include "NotificationPortHostTxReadyEvent.cpp"
```
typo? usually, we include headers instead of sources.

kcudnik committed:
[syncd] Add pre match logic for pfc storm (#1309)

Special case where priority is the same for all acl

noaOrMlnx committed:
Fix CR comments

noaOrMlnx committed:
Extend coverage for host_tx_ready enhancements

noaOrMlnx committed:
Update SAI community submodule pointer to latest master

noaOrMlnx committed:
Update on_port_host_tx_ready function in the right order and removed
whitespaces

jimmyzhai committed:
[gbsyncd] Add asic db prefix for channel RESTARTQUERY (#1302)

* Use different channel RESTARTQUERY for syncd/gbsyncd

* change restartquery to restartQuery due to aspell error

kcudnik committed:
[syncd] Translate removed RID on bfd session state change (#1320)

Fixes https://github.com/sonic-net/sonic-buildimage/issues/16992

noaOrMlnx committed:
Remove try-catch from json parse

Ashish1805 committed:
[Link Event Damping] Add link event damping sairedis port attributes. (#1314)

- Add link event damping sairedis port attributes.
- These attributes will be used by OA to set the link event damping
  algorithm and AIED link event damping config.
- Syncd uses these sairedis port attribute to receive the link event
  damping algorithm and AIED config to execute syncd-based link event
  damping.

HLD: sonic-net/SONiC#1071

noaOrMlnx committed:
Fix typo in UT

kcudnik committed:
[syncd] User fdb flush default value from SAI (#1317)

Previously we were hard-codding default value for fdb flush, and since
this value could change in the future, we want to get this form SAI
metadata

nazariig committed:
[hash]: Extend VS lib with ECMP/LAG hash algorithm. (#1316)

Signed-off-by: Nazarii Hnydyn nazariig@nvidia.com

HLD: sonic-net/SONiC#1501

Ashish1805 committed:
[Link Event Damping] Add Abstract SelectEventHandler class to handle (#1312)

Handler in the class will be used to handle the timer event of link event dampers.
HLD: sonic-net/SONiC#1071

Ashish1805 committed:
[Link Event Damping] Port state change handler class. (#1310)

- Class to handle the port state change event callback from SAI and
  sending the events to syncd main thread for processing by link event
  damper.

Depends on: sonic-net/sonic-sairedis#1297

HLD: sonic-net/SONiC#1071

dbarashinvd committed:
Add if statement for module control mode support (#1305)

* add if statement for independent module mode support

* change profile file name

Ashish1805 committed:
Fix compilation issue due to PORT_STATE_CHANGE_QUEUE_SIZE undefined (#1324)

In PR https://github.com/sonic-net/sonic-sairedis/pull/1310, a
  constexpt static data member was added. Since sairedis repo uses
  c++14, definition of this data member needs to be provided in .cpp file too.

kcudnik committed:
Merge branch 'master' into host_tx_ready_enhancements-sonic-sairedis
